infrastructure/k8s/nats.tf

```diff
@@ -0,0 +1,250 @@
+resource "kubernetes_namespace" "nats" {
+  metadata {
+    name = "nats"
+  }
+}
+
+resource "null_resource" "setup_nats" {
+  triggers = {
+    nats_namespace = kubernetes_namespace.nats.metadata[0].name
+  }
+
+  provisioner "local-exec" {
+    command = <<-EOT
+      helm repo add nats https://nats-io.github.io/k8s/helm/charts/
+      helm repo update
+      helm pull nats/nats --untar
+      helm template nats ./nats \
+        --namespace ${kubernetes_namespace.nats.metadata[0].name} \
+        --set cluster.enabled=true \
+        --set cluster.replicas=3 \
+        --set nats.jetstream.enabled=true > nats-resources.yaml
+      kubectl apply -f nats-resources.yaml
+    EOT
+  }
+}
+
+
+resource "null_resource" "create_natsschannel_crd" {
```

Comment:
Nitpick but maybe underscore between `nats` and `channel`

Reply:
gonna rename it anyway based on your above suggestion